lib/puppet.rb

```diff
@@ -36,6 +37,10 @@ module Puppet
   require 'puppet/environments'
 
   class << self
+    include FastGettext::Translation
+    GettextSetup.initialize(File.absolute_path('../locales', File.dirname(__FILE__)))
+    FastGettext.locale = GettextSetup.negotiate_locale(ENV["HTTP_ACCEPT_LANGUAGE"])
```

Comment:
`Encoding.default_external` might work:
```
└> ruby -e 'puts Encoding.default_external' 
UTF-8
└> LC_ALL=C ruby -e 'puts Encoding.default_external'
US-ASCII
```
That being said, if we're just using the system locale, FastGettext might do this under the hood.